lib/seahorn/VCGen.cc

```diff
@@ -146,6 +146,10 @@ void VCGen::genVcForCpEdge(OpSemContext &ctx, const CpEdge &edge) {
       checkSideAtBb(head, ctx.side(), bbV, *smt, edge, bb);
   }
 
+  // -- if the block is truncated, omit phi nodes
+  if (&*edge.rbegin() == &target.bb())
```

Comment:
why not add a boolean flag for this behaviour?

I've looked at this code several times. I'm sure it works for the examples you have. But I can't really understand the condition. Why not add `includePhi` boolean argument that is set to true by default, and set it to false when necessary?

Reply:
I see your point. My solution is unnecessarily complex.

How the code works: There is a structural difference between CPG edges, and the "truncated edges" I generate. The condition identifies "truncated edges".

I think that a flag would be better for the following reasons.
1. A flag is a more robust solution. It will work even if the implementation of CPG changes.
2. If I use a flag, then the intent of the code is more explicit.
3. If I use a flag, then the feature can be reused by code without "truncated edges".
I've updated `genVcForCpEdge` accordingly.